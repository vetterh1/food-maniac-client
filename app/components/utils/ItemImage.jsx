import React from 'react';
import MdLocalRestaurant from 'react-icons/lib/md/local-restaurant';

const ItemImage = props => {
  if (!props.id) return null;
  return (<object data={`/static/thumbnails/${props.id}.jpg`} type="image/jpg"><MdLocalRestaurant size={96} /></object>);
};


ItemImage.propTypes = {
  id: React.PropTypes.string,
};

export default ItemImage;
