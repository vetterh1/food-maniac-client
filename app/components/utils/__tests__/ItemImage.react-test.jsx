
import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
import ItemImage from '../ItemImage';

describe('ItemImage', () => {
  // Enzyme test... not finished :(
  // it('should render without throwing an error', () => {
  //   const itemImage = shallow(<ItemImage />);
  //   // expect(itemImage.find('object')).to.have.length(1);
  //   // expect(itemImage.find(MdLocalRestaurant)).to.have.length(1);
  //   // expect(shallow(<ItemImage />).contains(<div className="foo">Bar</div>)).toBe(true);
  // });

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <ItemImage />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match its snapshot with unknown pic (renders MdLocalRestaurant)', () => {
    const tree = renderer.create(
      <ItemImage id="idPic" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

/*
it('changes the class when hovered', () => {
  const component = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
*/