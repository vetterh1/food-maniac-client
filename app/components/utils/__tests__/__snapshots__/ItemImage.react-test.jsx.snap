// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemImage should match its empty snapshot 1`] = `null`;
